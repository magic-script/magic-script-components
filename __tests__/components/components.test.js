import React from "react";
import renderer from "react-test-renderer";

import {
  Slider,
  View,
  Toggle,
  ToggleGroup,
  TextEdit,
  Image,
  ScrollBar,
  ScrollView,
  ListViewItem,
  GridLayout,
  LinearLayout,
  DropdownListItem,
  Panel,
  Tab,
  PageView,
  WebView,
  PortalIcon,
  ColorPicker,
  TimePicker,
  DatePicker,
  CircleConfirmation,
  Content,
  Light,
  Line,
  Prism,
  Scene,
} from "../../src/components/components";

describe("Components", () => {
  describe("Slider", () => {
    test("should create component", () => {
      const component = renderer.create(<Slider />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("View", () => {
    test("should create component", () => {
      const component = renderer.create(<View />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("Toggle", () => {
    test("should create component", () => {
      const component = renderer.create(<Toggle />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("ToggleGroup", () => {
    test("should create component", () => {
      const component = renderer.create(<ToggleGroup />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("TextEdit", () => {
    test("should create component", () => {
      const component = renderer.create(<TextEdit />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("Image", () => {
    test("should create component", () => {
      const component = renderer.create(<Image />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("ScrollBar", () => {
    test("should create component", () => {
      const component = renderer.create(<ScrollBar />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("ScrollView", () => {
    test("should create component", () => {
      const component = renderer.create(<ScrollView />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("ListViewItem", () => {
    test("should create component", () => {
      const component = renderer.create(<ListViewItem />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("GridLayout", () => {
    test("should create component", () => {
      const component = renderer.create(<GridLayout />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("LinearLayout", () => {
    test("should create component", () => {
      const component = renderer.create(<LinearLayout />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("DropdownListItem", () => {
    test("should create component", () => {
      const component = renderer.create(<DropdownListItem />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("Panel", () => {
    test("should create component", () => {
      const component = renderer.create(<Panel />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("Tab", () => {
    test("should create component", () => {
      const component = renderer.create(<Tab />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("PageView", () => {
    test("should create component", () => {
      const component = renderer.create(<PageView />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("WebView", () => {
    test("should create component", () => {
      const component = renderer.create(<WebView />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("PortalIcon", () => {
    test("should create component", () => {
      const component = renderer.create(<PortalIcon />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("ColorPicker", () => {
    test("should create component", () => {
      const component = renderer.create(<ColorPicker />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("TimePicker", () => {
    test("should create component", () => {
      const component = renderer.create(<TimePicker />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("DatePicker", () => {
    test("should create component", () => {
      const component = renderer.create(<DatePicker />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("CircleConfirmation", () => {
    test("should create component", () => {
      const component = renderer.create(<CircleConfirmation />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("Content", () => {
    test("should create component", () => {
      const component = renderer.create(<Content />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("Light", () => {
    test("should create component", () => {
      const component = renderer.create(<Light />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("Line", () => {
    test("should create component", () => {
      const component = renderer.create(<Line />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("Prism", () => {
    test("should create component", () => {
      const component = renderer.create(<Prism />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("Scene", () => {
    test("should create component", () => {
      const component = renderer.create(<Scene />);
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });
  });
});
