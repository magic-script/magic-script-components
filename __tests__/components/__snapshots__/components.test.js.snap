// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components CircleConfirmation should create component 1`] = `<circleConfirmation />`;

exports[`Components ColorPicker should create component 1`] = `<colorPicker />`;

exports[`Components Content should create component 1`] = `<content />`;

exports[`Components DatePicker should create component 1`] = `<datePicker />`;

exports[`Components DropdownListItem should create component 1`] = `<dropdownListItem />`;

exports[`Components GridLayout should create component 1`] = `<gridLayout />`;

exports[`Components Image should create component 1`] = `<image />`;

exports[`Components Light should create component 1`] = `<light />`;

exports[`Components Line should create component 1`] = `<line />`;

exports[`Components LinearLayout should create component 1`] = `<linearLayout />`;

exports[`Components ListViewItem should create component 1`] = `<listViewItem />`;

exports[`Components PageView should create component 1`] = `<pageView />`;

exports[`Components Panel should create component 1`] = `<panel />`;

exports[`Components PortalIcon should create component 1`] = `<portalIcon />`;

exports[`Components Prism should create component 1`] = `<prism />`;

exports[`Components Scene should create component 1`] = `<scene />`;

exports[`Components ScrollBar should create component 1`] = `<scrollBar />`;

exports[`Components ScrollView should create component 1`] = `<scrollView />`;

exports[`Components Slider should create component 1`] = `<slider />`;

exports[`Components Tab should create component 1`] = `<tab />`;

exports[`Components TextEdit should create component 1`] = `<textEdit />`;

exports[`Components TimePicker should create component 1`] = `<timePicker />`;

exports[`Components Toggle should create component 1`] = `<toggle />`;

exports[`Components ToggleGroup should create component 1`] = `<toggleGroup />`;

exports[`Components View should create component 1`] = `<view />`;

exports[`Components WebView should create component 1`] = `<webView />`;
